server/resp/src/main/java/org/infinispan/server/resp/commands/json/JSONGET.java

```diff
@@ -0,0 +1,171 @@
+package org.infinispan.server.resp.commands.json;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.functional.FunctionalMap;
+import org.infinispan.functional.impl.FunctionalMapImpl;
+import org.infinispan.functional.impl.ReadOnlyMapImpl;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+import org.infinispan.server.resp.serialization.ResponseWriter;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.util.DefaultIndenter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter.Indenter;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * JSON.GET
+ *
+ * @see <a href="https://redis.io/commands/json.get/">JSON.GET</a>
+ * @since 15.1
+ */
+public class JSONGET extends RespCommand implements Resp3Command {
+   // Configure JsonPath to use Jackson. Path with missing final leaf will return
+   // null
+   // path with more the 1 level missing will rise exception
+   static Configuration config = Configuration.builder().options(Option.DEFAULT_PATH_LEAF_TO_NULL)
+         .options(Option.SUPPRESS_EXCEPTIONS)
+         .jsonProvider(new com.jayway.jsonpath.spi.json.JacksonJsonNodeJsonProvider())
+         .mappingProvider(new com.jayway.jsonpath.spi.mapper.JacksonMappingProvider())
+         .build();
+
+   public JSONGET() {
+      super("JSON.GET", -2, 1, 1, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler, ChannelHandlerContext ctx,
+         List<byte[]> arguments) {
+      byte[] key = arguments.get(0);
+      Args args;
+      try {
+         args = parseArgs(arguments);
+      } catch (Exception ex) {
+         handler.writer().wrongArgumentNumber(this);
+         return handler.myStage();
+      }
+
+      FunctionalMap.ReadOnlyMap<byte[], Object> cache = ReadOnlyMapImpl
+            .create(FunctionalMapImpl.create(handler.typedCache(null)));
```

Comment:
Does this not create a new cache for every invocation? Should we instead pass `RespServer.RESP_KEY_MEDIA_TYPE`?

Reply:
tbh it was not clear to me, copied from [here](https://github.com/infinispan/infinispan/blob/4461d97009d7b38916723fa7fb605f37510cb9fc/server/resp/src/main/java/org/infinispan/server/resp/commands/hll/PFADD.java#L35-L36).
but you're right it creates a new cache. I guess we should change to byte stream, I'm trying to do that but I'm fighting with the marshaller, something is missing in the JsonDoc.java protostream configuration

Reply:
seems that the only way to make it works is to use `handler.typedCache(null)`, every other setup makes the lookup of the JsonDoc marshaller to fail.
No idea, should we create an embedded cache infrastruct as for multimap (i.e. EmbeddedMultimapListCache) or using `typedCache(null)` as it is now is ok?
@jabolina any idea?